### PR TITLE
[Feature] New ecoRouter list

### PR DESCRIPTION
**Pull request by berteotti**
New selection list and router components on ecoRouter

Screenshot 2022-04-01 at 16 37 57

# To Test

1. Open the page `swap`
2. Choose a trading pair
- [ ] Verify if new selection list appears
- [ ] Verify there is a default option selected
- [ ] Check if there is a Route display (if trading pair has multiple routes)
3. Select a new option
- [ ] Selected route should change

### CODE REVIEW

**Comment by MilanVojnovic95** [comment]
Testet:
It looks like it overlaps with 'Routing Through' under the list.
image

On mobile version the scroll down arrows are shown across the list
image

The rest looks good, the only thing I noticed is that it is very slow when choosing options from the list or new pairs.

Verify new selection list appears ✔ 
Verify there is a default option selected ✔ 
Check if there is a Route display (if trading pair has multiple routes) ✔ 
Select a new option ✔

**Comment by berteotti** [comment]
> Testet: It looks like it overlaps with 'Routing Through' under the list. image
> 
> On mobile version the scroll down arrows are shown across the list image

This is now fixed. Can you check it please?

**Comment by niemam29** [comment]
Hey, i've noticed some problems with overlapping: 
cach2

**Comment by MilanVojnovic95** [comment]
> > Testet: It looks like it overlaps with 'Routing Through' under the list. image
> > On mobile version the scroll down arrows are shown across the list image
> 
> This is now fixed. Can you check it please?

Tested and now it looks ok.
These bugs from my comment have been fixed

**Comment by MilanVojnovic95** [comment]
Retested and looks ok to me 👍

File: src/components/swap/SwapPlatformSelector.tsx [hunk 1]
```diff
@@ -1,31 +1,38 @@
...
 import { useSwapState } from '../../state/swap/hooks'
 import { useGasFeesUSD } from '../../hooks/useGasFeesUSD'
-import { RowFixed } from '../Row'
-import { ONE_BIPS, PRICE_IMPACT_MEDIUM, ROUTABLE_PLATFORM_LOGO } from '../../constants'
-import styled from 'styled-components'
-import { computeSlippageAdjustedAmounts, computeTradePriceBreakdown, simpleWarningSeverity } from '../../utils/prices'
+import {
+  computeSlippageAdjustedAmounts,
+  computeTradePriceBreakdown,
+  limitNumberDecimalPlaces,
```
Comment (adamazad) [inline src/components/swap/SwapPlatformSelector.tsx:16]: Not sure though.

```suggestion
  limitNumberOfDecimalPlaces,
```

File: src/components/swap/SwapPlatformSelector.tsx [hunk 2]
```diff
@@ -1,31 +1,38 @@
...
-import { useIsMobileByMedia } from '../../hooks/useIsMobileByMedia'
+import {
+  SelectionListDetails,
+  SelectionListLabel,
+  SelectionListLabelWrapper,
+  SelectionListName,
+  SelectionListOption,
+  SelectionListReceiveAmount,
+  SelectionListWindowWrapper
+} from '../SelectionList'
```
Comment (adamazad) [inline src/components/swap/SwapPlatformSelector.tsx:30]: Lovely

File: src/components/swap/SwapPlatformSelector.tsx [hunk 3]
```diff
@@ -38,6 +45,24 @@ interface GasFeeProps {
   gasFeeUSD: CurrencyAmount | null
 }
 
+const StyledFlex = styled(Flex)`
+  gap: 4px;
+`
```
Comment (adamazad) [inline src/components/swap/SwapPlatformSelector.tsx:50]: Shouldn't `Flex` support gap prop? [`gap` is in the specs](https://css-tricks.com/almanac/properties/g/gap/)

```suggestion
const StyledFlexWithGap = styled(Flex)`
  gap: 4px;
`
```

File: src/components/swap/SwapPlatformSelector.tsx [hunk 4]
```diff
@@ -86,95 +112,109 @@ export function SwapPlatformSelector({
...
+          const slippageAdjustedAmounts = computeSlippageAdjustedAmounts(trade)
+          const tokenAmount = limitNumberDecimalPlaces(
+            isExactIn ? slippageAdjustedAmounts[Field.OUTPUT] : slippageAdjustedAmounts[Field.INPUT]
+          )
+
+          return (
+            <SelectionListOption
+              key={trade.platform.name}
+              isSelected={selectedTrade?.platform.name === trade.platform.name}
+              isLoading={false}
```
Comment (adamazad) [inline src/components/swap/SwapPlatformSelector.tsx:173]: Where else is this property computed? Perhaps we should remove it.

**Review by adamazad — APPROVED** [review]
Good job, Leo! Just mind the comments, and it's good. Also, review the function comments. There was some typo I couldn't point it.

File: src/components/swap/SwapPlatformSelector.tsx (hunk 1 above)
Comment (berteotti) [reply src/components/swap/SwapPlatformSelector.tsx:16]: I was thinking about "limit number's decimal places" but it also could be "limit decimal places" or "limit decimal places number". I liking "limitDecimalPlaces" but don't know if it's to vague

File: src/components/swap/SwapPlatformSelector.tsx (hunk 3 above)
Comment (berteotti) [reply src/components/swap/SwapPlatformSelector.tsx:50]: Rebass Flex doesn't have gap prop on its types...

File: src/components/swap/SwapPlatformSelector.tsx (hunk 4 above)
Comment (berteotti) [reply src/components/swap/SwapPlatformSelector.tsx:173]: This is being used in EcoBridge when waiting for option results :)

**Review by berteotti — COMMENTED** [review]
~~@adamazad just solved your comments. Can you give it a look please?~~ Just saw it was a approve, not a request changes :)

File: src/components/swap/SwapPlatformSelector.tsx (hunk 1 above)
Comment (berteotti) [reply src/components/swap/SwapPlatformSelector.tsx]: Implemented your suggestion.

**Review by denciu — APPROVED** [review]
Looks good, nice job!

**Pull request merged**